### DISCUSSION TITLE
How to handle multiple schema with code files?

### DISCUSSION THREAD

**Discussion by tyzAAA** [graphql-hive/graphql-eslint#1131] [Q&A]
Hi there, I am new to eslint, I am currently using graphql-eslint with our project,

However, we are writing **MULTIPLE** graphql clients (while of course they have **MULTIPLE** different servers/schemas) in our projects, and their related operations are **written in separated js/ts files.** 

How am I supposed to write my eslint config file to ensure each operation file (js/ts, gql`xxx`) are linted by their only related schemas? Should I rewrite the processors of graphql-eslint to generated virtual graphql files in separated locations?

**Comment by ek1z** [comment]
Rebooting this discussions with my own problems on the topic.

My question (or problem) is also concerning multiple endpoints that provide unique schemas, which have some overlapping definitions so merging them into one single is not an option. I'm running with version `4.0.0-alpha.0` and it seems like my problem is also described in the comments of the `graphql-config.ts` file:

https://github.com/dimaMachina/graphql-eslint/blob/41ebe49fb3290b4594ae3b296dec7a7314f36faf/packages/plugin/src/graphql-config.ts#L23-L27

So in affect when I run ESlint for the whole project (with the configuration I have described below) the config loader will omit the `graphQLConfig` for files in path `src/gql/b/operations/**/*.gql`, and it will use the first config it creates when ESLint is run, which is the config for `src/gql/a/a.schema.graphql` schema. This of course causes multiple `known-type-names` errors for the files under `b`.

The problem does not occur:
- if I run eslint against the files individually
- if I comment out the if-return from compiled `graphql-config.js` under `node_modules`

So perhaps then the question is am I doing something that is not supported by the lib?

The `.eslintrc.js` configuration I'm running is the following (***note: this does not currently work***):

```js
...
overrides: [
  {
    files: ['*.gql'],
    parser: '@graphql-eslint/eslint-plugin',
    plugins: ['@graphql-eslint'],
    rules: {
      '@graphql-eslint/alphabetize': [
        'error',
        {
          fields: ['ObjectTypeDefinition'],
          selections: ['OperationDefinition', 'FragmentDefinition'],
        },
      ],
      '@graphql-eslint/known-type-names': 'error',
    },
  },
  {
    files: ['src/gql/a/operations/**/*.gql'],
    parser: '@graphql-eslint/eslint-plugin',
    parserOptions: {
      graphQLConfig: {
        documents: ['src/gql/a/operations/**/*.gql'],
        schema: 'src/gql/a/a.schema.graphql',
        skipGraphQLConfig: true,
      },
    },
  },
  {
    files: ['src/gql/b/operations/**/*.gql'],
    parser: '@graphql-eslint/eslint-plugin',
    parserOptions: {
      graphQLConfig: {
        documents: ['src/gql/b/operations/**/*.gql'],
        schema: 'src/gql/b/b.schema.graphql',
        skipGraphQLConfig: true,
      },
    },
  }
]
...
```
